src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/JsonFileDeserializer.cs

```diff
@@ -16,7 +16,7 @@ private class DefaultJsonFileDeserializer : JsonFileDeserializer
         {
             public override TValue Deserialize<TValue>(string filePath) where TValue : class
             {
-                using var stream = new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.ReadWrite | FileShare.Delete);
+                using var stream = new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.Read);
```

Comment:
I think you'll still need `FileShare.ReadWrite | FileShare.Delete` to prevent things exploding due to file locks.